processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1010,6 +1024,17 @@ public QueryableIndex load(File inDir) throws IOException
         segmentBitmapSerdeFactory = new BitmapSerde.LegacyBitmapSerdeFactory();
       }
 
+      Map<String, Object> metadata = null;
+      ByteBuffer metadataBB = smooshedFiles.mapFile("metadata.drd");
+      if (metadataBB != null) {
+        metadata = mapper.readValue(
+            serializerUtils.readBytes(metadataBB, metadataBB.remaining()),
```

Comment:
null is a valid metadata value and hard failure should be OK because it will happen only if segment file somehow got corrupted.